### PR TITLE
Fix Node migration

### PR DESCRIPTION
**Pull request by cwdsuzhou**

### CODE REVIEW

File: pkg/controllers/gameservers/controller.go [hunk 1]
```diff
@@ -450,6 +456,16 @@ func (c *Controller) syncGameServerStartingState(gs *carrierv1alpha1.GameServer)
...
+		// update gs to failed
+		if len(gs.Status.Address) != 0 || len(gs.Status.NodeName) != 0 {
+			gs.Status.State = carrierv1alpha1.GameServerFailed
+			gs.Status.Conditions = append(gs.Status.Conditions, carrierv1alpha1.GameServerCondition{
+				Type:          "PodDeleted",
+				LastProbeTime: metav1.NewTime(time.Now()),
+				Message:       "Pod deleted",
+			})
+			return c.carrierClient.CarrierV1alpha1().GameServers(gs.Namespace).UpdateStatus(gs)
+		}
```
Comment (cwdsuzhou) [inline pkg/controllers/gameservers/controller.go:468]: If has  scheduled or is  deleted now, mark gs as Failed.

File: pkg/controllers/gameservers/controller.go [hunk 2]
```diff
@@ -437,9 +446,6 @@ func (c *Controller) syncGameServerStartingState(gs *carrierv1alpha1.GameServer)
 	if IsBeingDeleted(gs) {
 		return gs, nil
```
Comment (cwdsuzhou) [inline pkg/controllers/gameservers/controller.go:447]: BeingDeleted contains DeletionTimestamp != nil

File: pkg/controllers/gameservers/controller.go [hunk 3]
```diff
@@ -485,16 +501,14 @@ func (c *Controller) syncGameServerStartingState(gs *carrierv1alpha1.GameServer)
...
-// 2. Add pod nodeName and port to GameServer Status(if using host port)
-// 3. Check node name address changed
-// 4. Check pod status
+// 1. Add pod nodeName and port to GameServer Status(if using host port)
+// 2. Check node name address changed
+// 3. Check pod status
 func (c *Controller) syncGameServerRunningState(gs *carrierv1alpha1.GameServer) (*carrierv1alpha1.GameServer, error) {
 	klog.V(4).Infof("Start sync running state for: %v", gs.Name)
-	if !gs.DeletionTimestamp.IsZero() {
+	if IsBeingDeleted(gs) {
```
Comment (cwdsuzhou) [inline pkg/controllers/gameservers/controller.go:509]: Deleting, Failed, Exit return here

File: pkg/controllers/gameservers/controller.go [hunk 4]
```diff
@@ -627,28 +648,26 @@ func (c *Controller) getGameServerPod(gs *carrierv1alpha1.GameServer) (*corev1.P
...
 		updated = true
 		applyGameServerAddressAndPort(gs, pod)
-		return gs, updated, nil
+		return updated
 	}
-	address := pod.Status.PodIP
-	if pod.Spec.NodeName != gs.Status.NodeName || address != gs.Status.Address {
-		gs.Status.State = carrierv1alpha1.GameServerFailed
-		c.recorder.Event(gs, corev1.EventTypeWarning, string(gs.Status.State), "Node migration occurred")
-	}
```
Comment (cwdsuzhou) [inline pkg/controllers/gameservers/controller.go:641]: this will  not happen now, if pod deleted ,   gs will be deleted.

**Review by hustcat — DISMISSED** [review]
/lgtm

**Comment by hustcat** [comment]
CI failed

**Review by hustcat — APPROVED** [review]
/lgtm

**Pull request merged**